applications/FSIApplication/tests/FSI_problem_emulator_test.py

```diff
@@ -178,6 +186,144 @@ def RunTestCase(self):
+                "solver_type"             : "Dynamic",
+                "echo_level"              : 0,
+                "analysis_type"           : "linear",
+                "time_integration_method" : "implicit",
+                "scheme_type"             : "bossak",
+                "model_import_settings"              : {
+                    "input_type"       : "mdpa",
+                    "input_filename"   : "test_FSI_emulator_Structural",
+                    "input_file_label" : 0
+                },
+                "material_import_settings" :{
+                    "materials_filename": "materials_2D.json"
+                },
+                "line_search"                        : false,
+                "convergence_criterion"              : "Residual_criterion",
+                "displacement_relative_tolerance"    : 1e-8,
+                "displacement_absolute_tolerance"    : 1e-10,
+                "residual_relative_tolerance"        : 1e-8,
+                "residual_absolute_tolerance"        : 1e-10,
+                "max_iteration"                      : 20,
+                "rotation_dofs"                      : false,
+                "linear_solver_settings"             : {
+                    "solver_type" : "LinearSolversApplication.sparse_lu",
+                    "scaling"     : true
+                }
+            }
+        }
+        """)
+
+        with WorkFolderScope(self.work_folder):
+
+            self.model = Model()
+
+            # Construct the structure solver
+            self.structure_solver = python_solvers_wrapper_structural.CreateSolver(self.model, StructureSolverSettings)
+
+            self.structure_solver.AddVariables()
+
+            self.structure_solver.ImportModelPart()
+
+            self.structure_solver.PrepareModelPart()
+
+            self.structure_main_model_part = self.model["Structure"]
+
+            self.structure_solver.AddDofs()
+
+            self.SetStructureBoundaryConditions()
+
+            self.structure_solver.Initialize()
+
+            self.coupling_utility.Initialize()
+
+            residual_size = self.GetInterfaceProblemSize()*2 # Interface DOFs number times PROBLEM_SIZE
+            self.iteration_value_u = Vector(residual_size) # Interface displacement iteration guess
+            self.iteration_value_f = Vector(residual_size) # Interface fluxes iteration guess
+            self.corrected_iteration_value_u = Vector(residual_size) # Interface displacement iteration guess
+            self.corrected_iteration_value_f = Vector(residual_size) # Interface fluxes iteration guess
+            for i in range(0,residual_size):
+                self.iteration_value_u[i] = 0.0
+                self.iteration_value_f[i] = 0.0
+                self.corrected_iteration_value_u[i] = 0.0
+                self.corrected_iteration_value_f[i] = 0.0
+
+            step = 0
+            time = 0.0
+
+            while(time <= self.end_time):
```

Comment:
cannot this be further unified? => should be the same loop for al the accelerators no?

Reply:
Nope. If they are block Newton the current non-linear loop is not valid anymore (this is why I required to refactor all the solvers). What I'll do for the test upgrade is to use the FSI solver and create a fake fluid emulation solver with the springs.

Reply:
sure... 🤦